src/SocketheadCleanArch.API/Utils/AuthHelper.cs

```
@@ -0,0 +1,58 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text.Json;
+using Microsoft.IdentityModel.Tokens;
+using SocketheadCleanArch.Domain.Entities;
+
+namespace SocketheadCleanArch.API.Utils;
+
+public static class AuthHelper
+{
+    public static async Task<bool> ValidateAppleTokenSignatureAsync(string idToken)
+    {
+        using HttpClient httpClient = new();
+        HttpResponseMessage response = await httpClient.GetAsync("https://appleid.apple.com/auth/keys");
+        string content = await response.Content.ReadAsStringAsync();
+
+        JsonDocument appleKeys = JsonDocument.Parse(content);
+        JsonElement keys = appleKeys.RootElement.GetProperty("keys");
+
+        TokenValidationParameters validationParameters = new()
+        {
+            ValidateIssuer = true,
+            ValidIssuer = "https://appleid.apple.com",
+            ValidateAudience = false,
+            ValidateLifetime = true,
+            ValidateIssuerSigningKey = true,
+            IssuerSigningKeys = keys.EnumerateArray().Select(AppleKeyHelper.CreateSecurityKey)
+        };
+
+        JwtSecurityTokenHandler tokenHandler = new();
+        try
+        {
+            tokenHandler.ValidateToken(idToken, validationParameters, out _);
+            return true;
+        }
+        catch
+        {
```

We are swallowing this error.  Shouldn't we be logging here since this would be an unexpected exception (i.e. we got an invalid token somehow)?

---

Yes, you're right. I've added Serilog logger here and now handle the exception gracefully.